service/src/main/java/bio/terra/workspace/app/controller/ReferencedGcpResourceController.java

```diff
@@ -715,10 +718,10 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
   public ResponseEntity<ApiCloneReferencedGcpDataRepoSnapshotResourceResult>
       cloneGcpDataRepoSnapshotReference(
```

Comment:
Just curious, do you need the pet to check access to a TDR snapshot? We need it for the GCP resources because we're talking to the cloud directly, but I thought TDR wouldn't need additional OAuth scopes. Should be fine either way (user or pet) though.

Reply:
You're right, we don't need to use the pet to check snapshot access. IIRC snapshots are shared with proxy groups so we either the user or pet credentials should work, but the pet credentials aren't required.